### PR TITLE
Key routes added

### PR DESCRIPTION
**Pull request by monarkatfactly**

### CODE REVIEW

File: controllers/api/v1/users.js [hunk 1]
```diff
@@ -42,5 +42,5 @@ function getUserBySlug(req, res, next) {
 
 module.exports = function routes(router) {
     router.get('/', getUser);
```
Comment (ntallapa12) [inline controllers/api/v1/users.js]: Do you mind making all singular to plural (getUser to getUsers, getFactcheck to getFactchecks so on).

File: lib/validateClientHeader.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    return async function clientHeader(req, res, next) {
+        try {
+            if(!req.headers.client){
+                res.status(422).json({ error: {
+                    message: 'Client ID is missing',
+                    code: '500-001',
+                    requestId: req.headers['x-request-id'] || null
+                } });
+            }
+        } catch (e) {
```
Comment (ntallapa12) [inline lib/validateClientHeader.js]: catch(next) should take care of this, unless you want some explicit error handling

File: models/factcheck.js [hunk 3]
```diff
@@ -431,18 +431,8 @@ class FactcheckModel extends MongoBase {
...
-                            if(factchecks[i].claims){
-                                for(let j = 0; j < factchecks[i].claims.length; j++){
-                                    factchecks[i].claims[j].rating.media = ratingMediaObject[factchecks[i].claims[j].rating.media]
-                                }
-                            }
-                        }
-                        
-                        return factchecks
+                        //Traveling through all the factchecks and replacing DBref media with full media object                        
+                        return factchecks.map(factcheck => factcheck.claims && factcheck.claims.length > 0 ? { ...factcheck, claims: factcheck.claims.map(claim => claim.rating ? { ...claim, rating: {...claim.rating, media: ratingMediaObject[claim.rating.media]}} : claim) } : factcheck );
```
Comment (ntallapa12) [inline models/factcheck.js]: for inline programming add some comments (on 2 ternary operators) and indent for every .operation (in this case 2 .maps )

File: models/factcheck.js [hunk 4]
```diff
@@ -498,18 +488,8 @@ class FactcheckModel extends MongoBase {
...
-                        for(let i = 0; i < factchecks.length; i++){
-                            if(factchecks[i].claims){
-                                for(let j = 0; j < factchecks[i].claims.length; j++){
-                                    factchecks[i].claims[j].claimant.media = claimantMediaObject[factchecks[i].claims[j].claimant.media]
-                                }
-                            }
-                        }
-                        
-                        return factchecks
+                        return factchecks.map(factcheck => factcheck.claims && factcheck.claims.length > 0 ? { ...factcheck, claims: factcheck.claims.map(claim => claim.claimant ? { ...claim, claimant: {...claim.claimant, media: claimantMediaObject[claim.claimant.media]}} : claim) } : factcheck );
```
Comment (ntallapa12) [inline models/factcheck.js]: same as above

File: models/organization.js [hunk 5]
```diff
@@ -235,7 +233,9 @@ class OrganizationModel extends MongoBase {
             .aggregate(aggregations).toArray())
             .then((results) => {
                 this.logger.info('Retrieved the results');
-                return results;
+                return {
+                    data: results[0]
```
Comment (ntallapa12) [inline models/organization.js]: can there be a case for no orgs? it results in index not found error

File: lib/validateClientHeader.js (hunk 2 above)
Comment (monarkatfactly) [reply lib/validateClientHeader.js]: I didn't get this.

**Pull request merged**